wagtail/images/blocks.py

```diff
@@ -51,3 +57,156 @@ def htmldiff(self):
                 "image_b": self.val_b,
             },
         )
+
+
+class ImageBlock(StructBlock):
+    """
+    An usage of ImageChooserBlock with support for alt text.
+    For backward compatibility, this block overrides necessary methods to change
+    the StructValue to be an Image model instance, making it compatible in
+    places where ImageChooserBlock was used.
```

Comment:
Seems to me like there might be more work required to override the different methods of StructBlock to be compatible with a value provided as an Image model instance, based on Storm’s feedback here: https://github.com/wagtail/bakerydemo/pull/492#issuecomment-2314665023

For example, the way `get_searchable_content` is implemented, it would be provided with an image instance rather than a struct value. I worry other methods might require a similar treatment (usage counts?).

Reply:
The missing methods are now (hopefully) all implemented in #12416.